### ISSUE TITLE
Plan initialisation fails with initial model if input_shape or data.yaml entries are not provided

### ISSUE THREAD

**Issue by payalcha** [intel/openfl#1559]
**Describe the bug**
While plan initialisation with initial model below error is coming - 

**EXCEPTION** : 'Unable to construct full dataloader from collab_index=0 when the plan has 0 as total collaborator count. Please check plan/data.yaml file for current collaborator entries.'

**To Reproduce**

Create workspace 
fx workspace create --prefix my_workspace --template keras/mnist
Move into the folder 
cd mnist_example
Install dependencies 
pip install -r requirements.txt
Remove data.yaml entry (we don't want to download data)

Try to initialise the plan with input shape f
fx plan initialize -a localhost -i ../mnist_example/save/init.pbuf

**Expected behavior** - plan should be initialised successfully without the data download happening.

Actual result - fails with EXCEPTION : 'Unable to construct full dataloader from collab_index=0 when the plan has 0 as total collaborator count. Please check plan/data.yaml file for current collaborator entries.'

**Screenshots**

<img width="1440" alt="Image" src="https://github.com/user-attachments/assets/7a70f874-28cf-468d-a388-37283a6e2d37" />

**Note**
We intentionally deleted the data.yaml as our understanding is that if initial model is given to initialise, input-shape or data is not needed.

**Comment by kminhta** [comment]
Thanks for raising this issue. As per offline discussion, this (and issues related to `fx plan initalize`) should be addressed in a bigger clean-up / refactor. Higher level design around the refactoring is in progress @rahulga1 

Tagging https://github.com/securefederatedai/openfl/pull/1542#issuecomment-2827729065 as related effort. 

To add some context: the specific error we're seeing is caused because `fx plan initialize` is trying to initialize the `data_loader`, which requires `input_shape` or a seed data. In theory, loading a pbuf doesn't not actually need this. However, information about the input is still eventually needed even when a pretrained initial model is provided because the pretrained model only contains the model weights and not the actual model class or structure. What is happening is the model gets initialized via the task runner as it normally does, then the weights from the pretrained model are loaded in. Because of this, the model still needs the input shape to initialize.

**Comment by noopurintel** [comment]
Successfully tested - by removing the entries from `data.yaml` and using `init.pbuf` from another workspace

<img width="1312" alt="Image" src="https://github.com/user-attachments/assets/055ee3cc-dc75-4b0e-93bc-6f73293ace60" />

<img width="1312" alt="Image" src="https://github.com/user-attachments/assets/14d60459-fe2a-4950-9b55-b4f693981bc6" />
